src/internal/utils.ts

```diff
@@ -137,3 +137,15 @@ allVirtualoptions.push('ref');
 export function includesAllVirtualPOP(options: VirtualOptions): options is VirtualOptions {
   return allVirtualoptions.every((v) => Object.keys(options).includes(v));
 }
+
+/**
+ * Merges existing metadata with new value
+ * @param key Metadata key
+ * @param value Raw value
+ * @param cl The constructor
+ */
+export function assignMetadata(key: string, value: unknown, cl: new () => {}): void {
```

Comment:
I dont fully understand for what this is needed, could you explain?

Reply:
This is to ensure that you can use the deprecated Typegoose class and the annotation. 
To avoid overriding exiting metadata.

Reply:
Sorry, didnt see that this was used for the "settings" one, resolved